ansible/roles/host-ocp4-hcp-cnv-install/tasks/main.yaml

```diff
@@ -0,0 +1,304 @@
+
+    - name: Create the kubeconfig secret
+      kubernetes.core.k8s:
+        definition: "{{ lookup('ansible.builtin.template', 'kubeconfigsecret.yaml') }}"
+        wait: true
+        wait_timeout: 300
+      vars:
+        namespace: "{{ hcp_ocp_namespace }}"
+
+    - name: Create htpasswd secret for admin and userX
+      include_tasks: setup_htpasswd_authentication.yml
+
+    - name: Create the HostedCluster
+      kubernetes.core.k8s:
+        definition: "{{ lookup('ansible.builtin.template', 'hostedcluster.yaml') }}"
+        wait: true
+        wait_timeout: 300
+      vars:
+        namespace: "{{ hcp_ocp_namespace }}"
+
+    - name: Create the NodePool
+      kubernetes.core.k8s:
+        definition: "{{ lookup('ansible.builtin.template', 'nodepool.yaml') }}"
+        wait: true
+        wait_timeout: 300
+      vars:
+        namespace: "{{ hcp_ocp_namespace }}"
+
+    - name: Get HostedCluster
+      kubernetes.core.k8s_info:
+        api_version: hypershift.openshift.io/v1beta1
+        kind: HostedCluster
+        name: "hcp-{{ guid }}"
+        namespace: "{{ hcp_ocp_namespace }}"
+      register: r_hosted_cluster
+      retries: 120
+      delay: 10
+      until:
+        - r_hosted_cluster.resources[0].status.kubeconfig is defined
+        - r_hosted_cluster.resources[0].status.kubeconfig.name is defined
+        - r_hosted_cluster.resources[0].status.kubeconfig.name | length > 0
+
+    - name: Get the OpenShift admin kubeconfig secret
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Secret
+        name: hcp-{{ guid }}-admin-kubeconfig
+        namespace: "{{ hcp_ocp_namespace }}"
+      register: kubeadmin_secret_result
+
+    - name: Get the OpenShift secret
+      when: hcp_authentication | default('') != 'htpasswd'
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Secret
+        name: hcp-{{ guid }}-kubeadmin-password
+        namespace: "{{ hcp_ocp_namespace }}"
+      register: password_secret_result
+
+    - name: Decode the kubeconfig secret
+      set_fact:
+        kubeconfig_decoded: "{{ kubeadmin_secret_result.resources[0].data.kubeconfig | b64decode }}"
+
+    - name: Decode the kubeadmin password secret
+      when: hcp_authentication | default('') != 'htpasswd'
+      set_fact:
+        kubeadmin_password: "{{ password_secret_result.resources[0].data.password | b64decode }}"
+
+    - name: Generate /tmp/kubeconfig
+      copy:
+        dest: /tmp/kubeconfig
+        content: "{{ kubeconfig_decoded }}"
+
+- name: Get information about the router-nodeport-default service
+  kubernetes.core.k8s_info:
+    kubeconfig: /tmp/kubeconfig
+    api_version: v1
+    kind: Service
+    namespace: openshift-ingress
+    name: router-nodeport-default
+  register: service_info
+  retries: 10
+  delay: 30
+  until: service_info.resources[0].spec.ports is defined and service_info.resources[0].spec.ports | length > 0
```

Comment:
Just a comment: I usually write 'and' conditions as a list. But no preference really. :-)